src/Core/src/Hosting/IMauiServiceCollection.cs

```diff
@@ -1,11 +1,10 @@
 using System;
-using System.Collections.Generic;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Maui.Hosting
 {
-	public interface IMauiServiceCollection
-		: IServiceCollection, IDictionary<Type, Func<IServiceProvider, object?>?>
+	public interface IMauiServiceCollection : IServiceCollection
 	{
+		bool TryGetService(Type serviceType, out ServiceDescriptor? descriptor);
```

Comment:
The `ServiceDescriptor` is needed to actually learn if the service is a single ton or transient. No support yet for scoped.